conbench/runner.py

```diff
@@ -260,13 +260,16 @@ def record(self, result, name, error=None, publish=True, **kwargs):
         else:
             benchmark["machine_info"] = self.machine_info
 
-        run_name = options.get("run_name")
-        if run_name is not None:
-            benchmark["run_name"] = run_name
+        benchmark["run_name"] = options.get("run_name") or remaining_kwargs.pop('run_name', None)
+        if benchmark["run_name"] is None:
+            raise RuntimeError("conbench requires run_name to be set via --run-name CLI or passing run_name kwarg to `record()` func")
```

Comment:
I don't think this will actually impact anyone, but for context: the original reason these weren't marked as required like this is that really they are only required _if one is publishing results_. And the CLI run-conbench-manually-locally continues if there's a `POST` failure.

I don't think very many people are doing this run-manually (I know of one instance in the past few weeks, but very few-to-none other than that). The work that Ed's doing to separate the runners-bits out from the conbench bits will also totally change that interaction paradigm (and so we will avoid this issue entirely).

I'm fine to have these be errors, though maybe we should also update the docs to basically say "if you aren't sending results to conbench, make something up, it doesn't matter, but it has to be non-empty" https://github.com/conbench/conbench/blob/35fde9954e2b365e36472f16b46fe3067e9dc455/conbench/cli.py#L128-L141